File: src/main/java/org/devocative/keights/dto/RewriteRequest.java
package org.devocative.keights.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;

@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
@ToString
public class RewriteRequest {
	private EEventType event;
	private String domainName;
	private String serviceName;
	private String serviceNamespace;

	public RewriteRequest(String domainName, String serviceName, String serviceNamespace) {
		this.domainName = domainName;
		this.serviceName = serviceName;
		this.serviceNamespace = serviceNamespace;
	}

	public String toFQDN(String clusterDomain) {
		return String.format("%s.%s.svc.%s", serviceName, serviceNamespace, clusterDomain);
	}
}

File: src/main/kotlin/org/devocative/keights/dto/RewriteRequest.kt
package org.devocative.keights.dto

data class RewriteRequest(
    var event: EEventType,
    val domainName: String,
    val serviceName: String,
    val serviceNamespace: String
) {
    //var event: EEventType? = null

    fun toFQDN(clusterDomain: String) = "$serviceName.$serviceNamespace.svc.$clusterDomain"
}
